addons/web_editor/static/src/js/editor/odoo-editor/src/utils/utils.js

```diff
@@ -285,22 +285,24 @@ export function getFurthestUneditableParent(node, parentLimit) {
     return nonEditableElement;
 }
 /**
- * Returns the closest HTMLElement of the provided Node
- * if a 'selector' is provided, Returns the closest HTMLElement that match the selector
+ * Returns the closest HTMLElement of the provided Node. If a 'selector' is
+ * provided, returns the closest HTMLElement that match the selector. If a
+ * predicate is provided, the first closest that match the predicate will be
+ * returned. Any returned element will be contained within the editable.
  *
  * @param {Node} node
- * @param {string} [selector=undefined]
- * @returns {HTMLElement}
+ * @param {string} [selector='*']
+ * @param {string} [predicate=() => true]
+ * @returns {HTMLElement|null}
  */
-export function closestElement(node, selector) {
-    let element = node;
-    while (element && element.nodeType !== Node.ELEMENT_NODE) {
-        element = element.parentNode;
-    }
-    if (element && selector) {
-        element = element.closest(selector);
-    }
-    return element && element.querySelector('.odoo-editor-editable') ? null : element;
+export function closestElement(node, { selector = "*", predicate = () => true } = {}) {
```

Comment:
It feels so clumsy to have to write `{ selector: ...}` all the time we use this function, which is a lot. I suppose you'll disagree but personally I think I'd rather have a polymorphic parameter which is either a string or a function. In typescript it would be like `predicate: string | (node: Node) => Boolean`. This commit would become 10 lines instead of hundreds, and the enterprise PR could even be deleted.